pytest_bdd/parsers.py

```diff
@@ -0,0 +1,113 @@
+"""Step parsers."""
+
+from __future__ import absolute_import
+import re as base_re
+
+import parse as base_parse
+from parse_type import cfparse as base_cfparse
+
+import pytest
+import six
+
+from .exceptions import InvalidStepParserError
+
+
+RE_TYPE = type(base_re.compile(''))
+
+
+class StepArgumentParser(object):
+
+    """Parser of the individual step arguments."""
+
+    def __init__(self, name):
+        self.name = name
+
+    def get_params(self, name):
+        """Get step parameters from the given step name."""
+        raise NotImplementedError()
+
+    def is_matching(self, name):
+        """Match given name with the step name."""
+        raise NotImplementedError()
+
+
+class re(StepArgumentParser):
+
+    """Regex step parser."""
+
+    def __init__(self, name, **kwargs):
+        """Compile regex."""
+        super(re, self).__init__(name)
+        self.regex = base_re.compile(self.name, **kwargs)
+
+    def get_params(self, name):
+        """Get step parameters."""
+        return self.regex.match(name).groupdict()
+
+    def is_matching(self, name):
+        """Match given name with the step name."""
+        return bool(self.regex.match(name))
+
+
+class parse(StepArgumentParser):
+
+    """parse step parser."""
+
+    def __init__(self, name, **kwargs):
+        """Compile parse expression."""
+        super(parse, self).__init__(name)
+        self.parser = base_parse.compile(self.name, **kwargs)
+
+    def get_params(self, name):
```

Comment:
parse_arguments, to be precise

On 7 January 2015 at 15:30, Oleg Pidsadnyi <notifications@github.com> wrote:

> In pytest_bdd/parsers.py
> <https://github.com/olegpidsadnyi/pytest-bdd/pull/96#discussion-diff-22588184>
> :
>
> > +
> > +    def is_matching(self, name):
> > +        """Match given name with the step name."""
> > +        return bool(self.regex.match(name))
> > +
> > +
> > +class parse(StepArgumentParser):
> > +
> > +    """parse step parser."""
> > +
> > +    def __init__(self, name, **kwargs):
> > +        """Compile parse expression."""
> > +        super(parse, self).__init__(name)
> > +        self.parser = base_parse.compile(self.name, **kwargs)
> > +
> > +    def get_params(self, name):
>
> parse_params?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/olegpidsadnyi/pytest-bdd/pull/96/files#r22588184>.
>



-- 
Anatoly Bubenkov